include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -336,30 +336,31 @@ __parallel_transform_reduce(_ExecutionPolicy&& __exec, _Up __u, _Cp __combine, _
     __wgroup_size = ::std::min(__wgroup_size, oneapi::dpl::__internal::__kernel_work_group_size(
                                                   ::std::forward<_ExecutionPolicy>(__exec), __kernel));
 #endif
-    auto __mcu = oneapi::dpl::__internal::__max_compute_units(::std::forward<_ExecutionPolicy>(__exec));
-    auto __n_groups = (__n - 1) / __wgroup_size + 1;
-    __n_groups = ::std::min(decltype(__n_groups)(__mcu), __n_groups);
-    // TODO: try to change __n_groups with another formula for more perfect load balancing
+    // It can be replaced with another value for more perfect load balancing
+    auto __iters_per_witem = decltype(__wgroup_size)(4);
```

Comment:
Is there any reason to write the code with `decltype`. I understand what is written but to me it complicates readability.

Reply:
Just to not be dependent on actual type of `__wgroup_size`. What is your suggestion? 
```
using _WG_size =decltype(__wgroup_size);
_WG_size __iters_per_witem = 4;
```
?

Reply:
I basically mean that we have to much `auto` keywords that brings genericity in the place where (I am quite sure) it's not necessary. The code looks like

```cpp
std::size_t i = 0;
decltype(i) b = 5;
```
or similar.

I am not saying that we should rewrite the whole code (it might me ideal, though) but at least to create using somewhere outside of variable creation for the sake of readability. 

```cpp
using size_type = decltype(__wgroup_size);
size_t __iters_per_witem = 4;
```

or at least

```cpp
decltype(__wgroup_size) __iters_per_witem = 4;
```

So, yes. One of the options I proposed matches with yours. But I am still concerned with the number of `auto` usage times